src/streamlink/plugins/tvplayer.py

```diff
@@ -60,40 +47,36 @@ def authenticate(self, username, password):
         res = self.session.http.get(self.login_url)
         match = self.login_token_re.search(res.text)
         token = match and match.group(1)
-        res2 = self.session.http.post(self.login_url, data=dict(email=username, password=password, token=token),
+        res2 = self.session.http.post(self.login_url, data=dict(email=username, password=password, _token=token),
                          allow_redirects=False)
         # there is a 302 redirect on a successful login
         return res2.status_code == 302
 
-    def _get_stream_data(self, resource, channel_id, token, service=1):
-        # Get the context info (validation token and platform)
-        self.logger.debug("Getting stream information for resource={0}".format(resource))
-        context_res = self.session.http.get(self.context_url, params={"resource": resource,
-                                                         "gen": token})
-        context_data = self.session.http.json(context_res, schema=self.context_schema)
-        self.logger.debug("Context data: {0}", str(context_data))
-
-        # get the stream urls
-        res = self.session.http.post(self.api_url, data=dict(
-            service=service,
-            id=channel_id,
-            validate=context_data["validate"],
-            token=context_data.get("token"),
-            platform=context_data["platform"]["key"]),
-            raise_for_status=False)
-
-        return self.session.http.json(res, schema=self.stream_schema)
+    def _get_stream_data(self, expiry, key, token, uvid):
+        res = self.session.http.get(self.api_url + uvid,
+            params=dict(key=key, platform="chrome"),
+            headers={"Token": token,
+                     "Token-Expiry": expiry,
+                     "Uvid": uvid})
+
+        res_schema = self.session.http.json(res, schema=self.stream_schema)
+
+        if res_schema["response"]["stream"] == None:
+            res = self.session.http.get(self.stream_url,
+                params=dict(key=key),
+                headers={"Token": token,
+                         "Token-Expiry": expiry,
+                         "Uvid": uvid}).json()
```

Comment:
[PEP8: continuation line under-indented for visual indent](https://stackoverflow.com/a/15435837)

```suggestion
            res = self.session.http.get(
                self.stream_url,
                params=dict(key=key),
                headers={
                    "Token": token,
                    "Token-Expiry": expiry,
                    "Uvid": uvid
                }
            ).json()
```